examples/connext_dds/instance_statistics/c/instance_publisher.c

```diff
@@ -0,0 +1,249 @@
+        participant, &DDS_PUBLISHER_QOS_DEFAULT, NULL /* listener */,
+        DDS_STATUS_MASK_NONE);
+    if (publisher == NULL) {
+        fprintf(stderr, "create_publisher error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* Register type before creating topic */
+    type_name = instanceTypeSupport_get_type_name();
+    retcode = instanceTypeSupport_register_type(
+        participant, type_name);
+    if (retcode != DDS_RETCODE_OK) {
+        fprintf(stderr, "register_type error %d\n", retcode);
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* To customize topic QoS, use
+    the configuration file USER_QOS_PROFILES.xml */
+    topic = DDS_DomainParticipant_create_topic(
+        participant, "Example instance",
+        type_name, &DDS_TOPIC_QOS_DEFAULT, NULL /* listener */,
+        DDS_STATUS_MASK_NONE);
+    if (topic == NULL) {
+        fprintf(stderr, "create_topic error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* To customize data writer QoS, use
+    the configuration file USER_QOS_PROFILES.xml */
+    writer = DDS_Publisher_create_datawriter(
+        publisher, topic,
+        &DDS_DATAWRITER_QOS_DEFAULT, NULL /* listener */, DDS_STATUS_MASK_NONE);
+    if (writer == NULL) {
+        fprintf(stderr, "create_datawriter error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+    instance_writer = instanceDataWriter_narrow(writer);
+    if (instance_writer == NULL) {
+        fprintf(stderr, "DataWriter narrow error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* Create data sample for writing */
+    instance = instanceTypeSupport_create_data_ex(DDS_BOOLEAN_TRUE);
+    if (instance == NULL) {
+        fprintf(stderr, "instanceTypeSupport_create_data error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* Main loop */
+    for (count=0; (sample_count == 0) || (count < sample_count); ++count) {
+
+        printf("Writing instance, count %d\n", count);
+
+        /* Modify the data to be written here */
+        instance->x = count;
+        /* register the keyed instance prior to writing */
+        instance_handle = instanceDataWriter_register_instance(
+                instance_writer,
+                instance);
+        /* Write data */
+        retcode = instanceDataWriter_write(
+            instance_writer, instance, &instance_handle);
+        if (retcode != DDS_RETCODE_OK) {
+            fprintf(stderr, "write error %d\n", retcode);
+        }
+
+        NDDS_Utility_sleep(&send_period);
+        if (count % 2 == 0) {
+            /* Unregister the instance */
+            retcode = instanceDataWriter_unregister_instance(
+                    instance_writer, instance, &instance_handle);
+            if (retcode != DDS_RETCODE_OK) {
+                fprintf(stderr, "unregister instance error %d\n", retcode);
```

Comment:
I do not know if we should continue if we fail to unregister or dispose

Reply:
already answer